### ISSUE TITLE
Improved development and production workflows

### ISSUE THREAD

**Issue by mhamann** [llama-farm/llamafarm#330]
### Goal
A user can easily spin up a llamafarm project with minimal external dependencies and then package it for other developers or for production, which may be on a different platform.

### The problem
Today, llamafarm requires the following system dependencies before most commands will work:
- llamafarm cli
- docker
- ollama / lemonade

While this works fairly well, it's a bit cumbersome to set up. If a user doesn't have docker, they need to install that first. If they don't have ollama, lemonade, or some other external runtime that llamafarm supports, they need to install that as well (most are likely in this second boat).

Due to the complexity of running some models, even this setup might not be enough. For example, some multimodal use cases (like images) aren't available on llama.cpp-based runtimes or don't work as well. And, if there's a hope of doing fine-tuning and training using the llamafarm dev kit, the ability to run pytorch and transformers with full hardware acceleration (even on macos) will be needed.

### Solution: development
For development, run everything natively on the user systems without leveraging docker. It would work something like this:
- Install llamafarm-cli as `lf`
- `lf` grabs a shallow copy of the llamafarm codebase that matches the current `lf version`
- `lf` installs `uv` (tiny, simple binary)
- `lf` invokes `uv` to install a llamafarm-specific version of python and runs `uv sync` to pull all dependencies
- `lf` starts the necessary backend processes as needed using `uv`

**Sharing a project**
When a user wants to share a project to a team mate, including all rag data, model customizations, etc, they can run `lf pack dev` to get a development archive that they can send to anyone. If the data isn't important to them, they can always just send `llamafarm.yaml` or even their project folder via `git`.

### Solution: production
We'll assume that true "production-like" deployments are using a Linux environment with an `amd64` architecture (`arm64` seems to lack popularity in this space for now, but it'll probably get more popular at some point).

Because things like `nvidia-container-toolkit` exists and Docker runs *natively* on Linux, llamafarm can run entirely in containers. So, a production flow might look like this...

A developer or CI/CD pipeline runs `lf pack prod` to get a production archive. This includes RAG databases, customized models, etc. Likely, some flags will be available here to change behaviors like: including all models, container archives, etc. An `lf deploy` command might also be provided to produce artifacts for Kubernetes, push stuff to Cloudflare, or even SSH into a remote machine and remotely install everything.

A new, optional section of `llamafarm.yaml` can be used to configure pack/deploy targets for specific architectures and environments:

```yaml
envs:
  dev:
    targets:
        - type: macos
    include:
      - models:custom
      - rag
  prod:
      targets:
        - type: linux
           gpus: ['cuda', 'rocm']
      include:
        - models
        - containers
        - rag
```

#### What this gives us
1) Users don't need to worry about dependencies for dev. It's as simple as possible.
2) LlamaFarm can take advantage of hardware acceleration (especially on Mac) where Docker doesn't currently provide good support.
3) LlamaFarm can better manage its own models and take advantage of existing models on the system (e.g., via huggingface).

#### Open questions
- How do we switch seamlessly between MLX/Diffusers and ONNX/ORT/TensorRT?
- How do we deploy/bake things for CUDA vs ROCm, etc?
